ext/shmop/php_shmop.stub.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+/** @return resource|false */
+function shmop_open(int $key, string $flags, int $mode, int $size) {}
+
+/**
+ * @param resource $shmid
+ * @return string|false
+ */
+function shmop_read($shmid, int $start, int $count) {}
+
+/**
+ * @param resource $shmid
+ */
+function shmop_close($shmid): bool {}
```

Comment:
Return type should be `void`.

Reply:
Does failure not result in `false` being returned? https://github.com/php/php-src/blob/de4e353c50ee62afe7c2ba5143b1349f64d307bc/ext/shmop/shmop.c#L276

Reply:
Same as with zip_close(). If zend_fetch_resource() wouldn't throw, the function's return type would be `?false` (which actually is the case for PHP < 8 – isn't that nice?)